### ISSUE TITLE
Update scalaz version to 7.2.0

### ISSUE THREAD

**Issue by gwils** [etorreborre/specs2#444]

**Comment by etorreborre** [comment]
I will do this with the next release. In the meantime you can grab https://oss.sonatype.org/content/repositories/releases/org/specs2/specs2-core_2.11/3.6.6-scalaz-7.2.0

**Comment by gwils** [comment]
Thanks @etorreborre!

**Comment by karlroberts** [comment]
Hi Eric, 
I had to remove my dependency on scalaz-7.2.0 and instead depend on the lib you specified above or else I get this error:-

    [error] Could not run test my.FooSpec: java.lang.IncompatibleClassChangeError: Found class scalaz.Memo, but interface was expected
    [error] Could not run test my.BarSpec: java.lang.NoClassDefFoundError: Could not initialize class org.specs2.main.Arguments$

Is this temporary until you push the next release or should we always depend on your version of scalaz when using specs2-3.3.6 ?

Also why does this happen is it binary incompatible? 
What about if I wanted to use Typelevel's Cats instead of Scalaz?

Cheers

Karl

**Comment by etorreborre** [comment]
@karlroberts if you want to use Scalaz 7.2.0 you need to use a specific version of specs2 (the one above). That's because Scalaz 7.1 and 7.2 are not binary compatible. In the next release, 3.6.7, I will probably switch to Scalaz 7.2 being the default and also publish a `specs2-3.6.7-scalaz-7.1`.

You can definitely use cats alongside the scalaz dependency that specs2 brings with it, that shouldn't be a problem.

**Comment by etorreborre** [comment]
This is done now with `specs2 3.7`.

**Comment by faucct** [comment]
You probably should fix the scalaz version dependency as `<7.2.3` then. Running tests with scalaz 7.2.3 results in those confusing exceptions:
```
java.lang.ClassCastException: scalaz.Free$Gosub cannot be cast to scala.Tuple2
        at scalaz.effect.IO$class.unsafePerformIO(IO.scala:22)
        at scalaz.effect.IOFunctions$$anon$6.unsafePerformIO(IO.scala:227)
        at org.specs2.runner.SbtRunner$$anonfun$newTask$1$$anon$4.execute(SbtRunner.scala:40)
        at sbt.TestRunner.runTest$1(TestFramework.scala:76)
        at sbt.TestRunner.run(TestFramework.scala:85)
        at sbt.TestFramework$$anon$2$$anonfun$$init$$1$$anonfun$apply$8.apply(TestFramework.scala:202)
        at sbt.TestFramework$$anon$2$$anonfun$$init$$1$$anonfun$apply$8.apply(TestFramework.scala:202)
        at sbt.TestFramework$.sbt$TestFramework$$withContextLoader(TestFramework.scala:185)
        at sbt.TestFramework$$anon$2$$anonfun$$init$$1.apply(TestFramework.scala:202)
        at sbt.TestFramework$$anon$2$$anonfun$$init$$1.apply(TestFramework.scala:202)
        at sbt.TestFunction.apply(TestFramework.scala:207)
        at sbt.Tests$$anonfun$9.apply(Tests.scala:216)
        at sbt.Tests$$anonfun$9.apply(Tests.scala:216)
        at sbt.std.Transform$$anon$3$$anonfun$apply$2.apply(System.scala:44)
        at sbt.std.Transform$$anon$3$$anonfun$apply$2.apply(System.scala:44)
        at sbt.std.Transform$$anon$4.work(System.scala:63)
        at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:226)
        at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:226)
        at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
        at sbt.Execute.work(Execute.scala:235)
        at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:226)
        at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:226)
        at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:159)
        at sbt.CompletionService$$anon$2.call(CompletionService.scala:28)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)
```

**Comment by etorreborre** [comment]
You can try [specs2-3.8.3](https://oss.sonatype.org/content/repositories/releases/org/specs2/specs2-common_2.11/3.8.3/specs2-common_2.11-3.8.3.pom). It depends on Scalaz 7.2.2 and should be compatible with 7.2.3.
